config/standard/test/tasks.spec.js

```diff
@@ -421,10 +421,10 @@ describe('Standard Configuration Tasks', () => {
         if (deliveryTaskDays.includes(day)) {
           it('should have "pregnancy-missing-birth" visit task', () => {
             // given
-            harness.pushMockedReport(backdatedReport('p', day));
+            harness.pushMockedDoc(backdatedReport('p', day));
 
             // when
-            return harness.getTasks({ resolved: true })
+            return harness.getTasks()
```

Comment:
This change made

```
      AssertionError: expected 0 to equal 1
      + expected - actual

      -0
      +1
```

at line 431,
```
assert.equal(tasks.length, 1);
```

So it stopped returning values

Reply:
Does it return results if the the task is not resolved? So you'd assert the tasks before calling `pushMockedDoc`, and then run `pushMockedDoc` and assert that the task has been resolved?

Reply:
you're upgrading to cht-conf-test-harness@2.x which is a breaking change and is only supported for projects after cht-core v3.8.    check out this history of the behaviour of tasks prior to cht-core v3.8 as it is relevant here.

> Once a task appears (appliesIf: true, resolvedIf: false), that task will not disappear until it is explicitly told to resolve (appliesIf: true, resolvedIf: true).

So when these tests were written, it was actually useful to check that a task was not only "not there" but that it was actually "resolved properly".  After 3.8, this assertion doesn't have much value. A task which has `appliesIf: false` behaves the same as a task with `resolvedIf: false`.  The only difference is completion vs cancelation. 

I'd just replace the whole `then` block with `assert.equal(tasks.length, 0);` which confirms that the task is not visible. I don't really understand the value from the other assertions. cc @eljhkrr who wrote it and may have more

Reply:
Reverted standard config and it's passing tests.

Reply:
This test is ancient and needs an update. Rather than mock both reports and check for resolved tasks, it should be sequential according to the workflow:
- Mock delivery report, assert that pnc task is visible
- Mock PNC, assert that no tasks are visible i.e. task has been resolved

Ideally actual forms should be filled out using harness.fillForm instead of mocking reports so that we can be sure everything works well from the users perspective